PennMobile/src/main/AndroidManifest.xml

```diff
@@ -5,6 +5,22 @@
     android:versionName="3.1.4">
 
     <uses-permission android:name="android.permission.WAKE_LOCK" />
+
+    <uses-feature
+        android:glEsVersion="0x00020000"
+        android:required="true" />
+
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" /> <!-- for Penn Transit API -->
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="com.google.android.providers.gsf.permission.READ_GSERVICES" /> <!-- vibration for laundry alert -->
+    <uses-permission android:name="com.pennapps.labs.pennmobile.permission.MAPS_RECEIVE" /> <!-- adding new contacts -->
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+    <uses-permission android:name="android.permission.VIBRATE" />
+    <uses-permission android:name="android.permission.READ_CONTACTS" />
+    <uses-permission android:name="android.permission.WRITE_CONTACTS" />
+
```

Comment:
Currently deleted Read and Write contacts since they are super sus and have danger protection level, but haven't tested as the dining hall server is down.  Would resolve after later testing.